### PR TITLE
fixed incorrect doctags in most_similar #601

### PR DESCRIPTION
**Pull request by parulsethi**
most_similar() in doc2vec.py doesn't consider given clip range for tagID/indexes [#601](https://github.com/RaRe-Technologies/gensim/issues/601)
fix: add clip_start to index_to_doctag's indexes.

### CODE REVIEW

**Comment by tmylk** [comment]
Thanks for the fix.
Please add a line in CHANGELOG and a new test for this.

File: gensim/models/doc2vec.py [hunk 1]
```diff
@@ -460,7 +460,7 @@ def most_similar(self, positive=[], negative=[], topn=10, clip_start=0, clip_end
             return dists
         best = matutils.argsort(dists, topn=topn + len(all_docs), reverse=True)
         # ignore (don't return) docs from the input
-        result = [(self.index_to_doctag(sim), float(dists[sim])) for sim in best if sim not in all_docs]
+        result = [(self.index_to_doctag(sim + clip_start), float(dists[sim])) for sim in best if (sim+clip_start) not in all_docs]
```
Comment (piskvorky) [inline gensim/models/doc2vec.py]: PEP8: space around binary operators.

**Comment by parulsethi** [comment]
Made the changes mentioned above.
Though there was an unrelated test failure once in test_word2vec [here](https://travis-ci.org/RaRe-Technologies/gensim/jobs/174344375).

File: gensim/test/test_doc2vec.py [hunk 2]
```diff
@@ -159,6 +160,12 @@ def model_sanity(self, model):
         self.assertEqual(list(zip(*sims))[0], list(zip(*sims2))[0])  # same doc ids
         self.assertTrue(np.allclose(list(zip(*sims))[1], list(zip(*sims2))[1]))  # close-enough dists
 
+        # sim results should be in clip range if given
+        clip_sims = model.docvecs.most_similar(fire1, clip_start=len(model.docvecs)//2, clip_end=len(model.docvecs))
```
Comment (tmylk) [inline gensim/test/test_doc2vec.py]: Add a clip_end as `len(model.docvecs)*2//3`

**Comment by tmylk** [comment]
Thanks for the PR!

**Pull request merged**